src/hotspot/share/services/memTracker.cpp

```diff
@@ -158,9 +126,8 @@ void MemTracker::final_report(outputStream* output) {
   // the final report again. In addition, it should be guarded from
   // recursive calls in case NMT reporting itself crashes.
   if (Atomic::cmpxchg(&g_final_report_did_run, false, true) == false) {
-    NMT_TrackingLevel level = tracking_level();
-    if (level >= NMT_summary) {
-      report(level == NMT_summary, output, 1);
+    if (enabled()) {
```

Comment:
Looks to me, you can do `enabled() && Atomic::cmpxchg(...)`, thus saving a tiny bit of CPU cycles when NMT is disabled?